src/layoutparser/models/paddledetection/layoutmodel.py

```diff
@@ -0,0 +1,272 @@
+from .catalog import PathManager, LABEL_MAP_CATALOG
+from ..base_layoutmodel import BaseLayoutModel
+from ...elements import Rectangle, TextBlock, Layout
+
+
+__all__ = ["PaddleDetectionLayoutModel"]
+
+
+
+class PaddleDetectionLayoutModel(BaseLayoutModel):
+    """
+    Args:
+        config (object):
+            config of model, defined by `Config(model_dir)`
+        model_path (str):
+            The path to the saved weights of the model.
+        threshold (float):
+            threshold to reserve the result for output
+        input_shape(list):
+            the image shape after reshape
+        batch_size(int)：
+            test batch size
+        label_map (:obj:`dict`, optional):
+            The map from the model prediction (ids) to realword labels (strings).
+        enforce_cpu (bool):
+            whether use cpu, if false, indicates use GPU
+        enable_mkldnn(bool):
+            whether use mkldnn to accelerate the computation
+        thread_num(int):
+            the number of threads
+    Examples::
+        >>> import layoutparser as lp
+        >>> model = lp.models.PaddleDetectionLayoutModel('
+                                    lp://PubLayNet/ppyolov2_r50vd_dcn_365e_publaynet/config')
+        >>> model.detect(image)
+    """
+    DEPENDENCIES = ["paddlepaddle"]
+    MODULES = [
+        {
+            "import_name": "_inference",
+            "module_path": "paddle.inference",
+        },
+    ]
+    DETECTOR_NAME = "paddledetection"
+
+    def __init__(self,
+                 config_path=None,
+                 model_path=None,
+                 label_map=None,
+                 enforce_cpu=False,
+                 extra_config={}):
+        if config_path is not None and config_path.startswith("lp://"):
+            prefix = "lp://"
+            if label_map is None:
+                dataset_name = config_path.lstrip("lp://").split("/")[0]
+                label_map = LABEL_MAP_CATALOG[dataset_name]
+            model_name = config_path[len(prefix) :].split('/')[1]
+            config_path = self._reconstruct_path_with_detector_name(config_path)
+            model_tar = PathManager.get_local_path(config_path)
+
+            pre_dir = os.path.dirname(model_tar)
+            base_dir = os.path.splitext(os.path.basename(model_tar))[0]
+            model_dir = os.path.join(pre_dir, base_dir)
+            self.untar_files(model_tar, model_dir)
+        if model_path is not None:
+            model_dir = model_path
+        self.predictor = self.load_predictor(
+            model_dir,
+            batch_size=extra_config.get('batch_size',1),
+            enforce_cpu=enforce_cpu,
+            enable_mkldnn=extra_config.get('enable_mkldnn',True),
+            thread_num=extra_config.get('thread_num',10))
+
+        self.threshold = extra_config.get('threshold',0.5)
+        self.input_shape = extra_config.get('input_shape',[3,640,640])
+        self.label_map = label_map
+        self.im_info = {
```

Comment:
So basically this is something like `default_image_preprocess_parameters`, right? It's better to use a more descriptive name rather than `im_info`, which is also the outputs of the preprocessors.